### PR TITLE
Add support for Group.BarrierPopCount on OpenCL

### PR DESCRIPTION
**Pull request by m4rs-mt**
Using `Group.BarrierPopCount` in OpenCL throws an `InternalCompilerError`, since this function cannot be mapped to a 'native' OpenCL instruction directly. This PR adds a new intrinsic software implementation for OpenCL to implement this function (see also #98).

### CODE REVIEW

File: Src/ILGPU/IR/Transformations/Optimizer.cs [hunk 1]
```diff
@@ -124,6 +124,11 @@ public static class Optimizer
             // Apply final DCE phase in release mode
             if (level > OptimizationLevel.O0)
                 builder.Add(new DeadCodeElimination());
+
+            // Append further backend specific transformations in release mode
```
Comment (MoFtZ) [inline Src/ILGPU/IR/Transformations/Optimizer.cs:128]: Is this change intented for this PR? It appears unrelated to Group.PopBarrierCount

Comment (m4rs-mt) [reply Src/ILGPU/IR/Transformations/Optimizer.cs:128]: Yes and no; perhaps this change should have been submitted in aother PR However, the OpenCL intrinsic does not appear to compile if the newly added `Group.BarrierPopCount` implementation is not inlined in the context of several AMD drivers. For this reason, I have modified the default backend transformations to include the `Inliner` phase by default (which was previously only enabled for the `PTXBackend`).

Comment (MoFtZ) [reply Src/ILGPU/IR/Transformations/Optimizer.cs:128]: Ah, I see. Fair enough.
